BUG: 32-bit wheels repo test failure in optimize

@mdhaber this may already be dealt with in `master` but I don't see an obvious PR at the moment? Not sure myself, but is blocking SciPy `1.6.1` release. Looks like very small tolerance issue and only 32-bit + Python 3.7; Azure logs disappear after a few days but: https://dev.azure.com/numpy/numpy/_build/results?buildId=15520&view=logs&j=e9afaa34-1a0f-534e-78fc-fae528ccd915&t=ba794874-3e80-5cdf-26c6-8142c057b516

and see below

```
......                                                                   [100%]
=================================== FAILURES ===================================
_________________ TestApproxDerivativeSparse.test_equivalence __________________
[gw0] linux -- Python 3.7.6 /venv/bin/python
/venv/lib/python3.7/site-packages/scipy/optimize/tests/test__numdiff.py:574: in test_equivalence
    assert_equal(J_dense, J_sparse.toarray())
E   AssertionError: 
E   Arrays are not equal
E   
E   Mismatched elements: 1 / 2500 (0.04%)
E   Max absolute difference: 3.55271368e-15
E   Max relative difference: 1.79430002e-16
E    x: array([[  0.4     ,   0.24    ,  -0.      , ...,   0.      ,  -0.      ,
E             0.      ],
E          [  0.6     ,  -0.44    ,   0.54    , ...,   0.      ,  -0.      ,...
E    y: array([[  0.4     ,   0.24    ,   0.      , ...,   0.      ,   0.      ,
E             0.      ],
E          [  0.6     ,  -0.44    ,   0.54    , ...,   0.      ,   0.      ,...
        J_dense    = array([[  0.39999996,   0.23999995,  -0.        , ...,   0.        ,
         -0.        ,   0.        ],
       [  0....9.99997   ],
       [ -0.        ,   0.        ,  -0.        , ...,   0.        ,
         29.39999706, 224.999955  ]])
        J_sparse   = <50x50 sparse matrix of type '<class 'numpy.float64'>'
	with 2500 stored elements in Compressed Sparse Row format>
        groups     = array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49])
        method     = 'cs'
        self       = <scipy.optimize.tests.test__numdiff.TestApproxDerivativeSparse object at 0xdd822bac>
        structure  = array([[1, 1, 1, ..., 1, 1, 1],
       [1, 1, 1, ..., 1, 1, 1],
       [1, 1, 1, ..., 1, 1, 1],
       ...,
       [1, 1, 1, ..., 1, 1, 1],
       [1, 1, 1, ..., 1, 1, 1],
       [1, 1, 1, ..., 1, 1, 1]])

```

--- a/scipy/optimize/tests/test__numdiff.py
+++ b/scipy/optimize/tests/test__numdiff.py
@@ -571,7 +571,8 @@ class TestApproxDerivativeSparse(object):
             J_dense = approx_derivative(self.fun, self.x0, method=method)
             J_sparse = approx_derivative(
                 self.fun, self.x0, sparsity=(structure, groups), method=method)
-            assert_equal(J_dense, J_sparse.toarray())
+            assert_allclose(J_dense, J_sparse.toarray(),
+                            rtol=5e-16, atol=7e-15)
 
     def test_check_derivative(self):
         def jac(x):

